### DISCUSSION TITLE
Shard2Shard Storage Policy?

### DISCUSSION THREAD

**Discussion by sv3t0sl4v** [ClickHouse/ClickHouse#33477] [Q&A] [answered]
OK. Storage policies are great. Faster volume on front for hot data, slower storage for cold data and all that... What if my shards have different storage speed? Let's say shard 1 is of 3 replicas each with 25 SSD (hot volume on server) and 25 SFF SAS-2 HDD 10k RPM (cold volume on modular storage array - SAS cable attached to second RAID card) storage and my shard 2 is a node of slower LFF HDD volume with incredibly higher capacity. Storage policies could move data from hot to cold disks, **but within a node**. Could they do it from shard to shard, so as data becomes older and older it gets pushed from **hot** to **cold** then to "**colder**" storage on another shard. which has even slower volume, bur larger capacity? Any advice on how to approach sharding out for larger storage capacity as data grows? Not into S3, really. Would prefer the benefits of optical LAN between nodes instead. What if I add **hotter** storage of NVMe SSDs on a third RAID card on each node in shard1? This is getting too weird, because **weight** for shards does not make much sense to me, if I want it to operate in this way described above as weight makes data to be distributed across shards in the amount proportional to the shard weigh, so it kind of implies shards should have same storage speed. What am I missing here?
Maybe it is better to use something like `TTL date_column + INTERVAL 1 WEEK TO VOLUME 'shard2_server_vol'` where shard2_server_vol is a volume accessed over the LAN on the other server, which was supposed to be used for shard2. but in this case just used as storage server+ modular storage arrays for larger capacity volume? Oh...

**Comment by tsolodov** [comment, marked as answer]
Hi @sv3t0sl4v , you should get some ideas from [this article](https://clickhouse.com/docs/en/guides/sre/scaling-clusters/)


_ClickHouse does not support automatic shard rebalancing. However, there are ways to rebalance shards in order of preference:_

_If you need to rebalance existing data and you have partitioned your data, consider detaching partitions and manually relocating them to another node before reattaching to the new shard. This is more manual than subsequent techniques but may be faster and less resource-intensive. This is a manual operation and thus needs to consider the rebalancing of the data._
